### PR TITLE
Enable the no tiered compilation innerloop PR legs

### PR DESCRIPTION
**Pull request by noahfalk**
Also using this PR to manually trigger test jobs. Marking no merge until manual testing is complete.

### CODE REVIEW

**Comment by noahfalk** [comment]
@dotnet-bot help

**Comment by noahfalk** [comment]
@dotnet-bot test Windows_NT x64 Checked Build and Test (Jit - TieredCompilation=0 )
@dotnet-bot test Windows_NT x86 Checked Build and Test (Jit - TieredCompilation=0 )
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test

**Comment by noahfalk** [comment]
@dotnet-bot test Windows_NT x64 Checked no_tiered_compilation_innerloop
@dotnet-bot test Windows_NT x86 Checked no_tiered_compilation_innerloop

**Comment by noahfalk** [comment]
The new tiered compilation jitstress1 legs
@dotnet-bot test Ubuntu arm64 Cross Checked r2r_jitstress1_tiered Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked jitstress1_tiered Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked r2r_jitstress1_tiered Build and Test
@dotnet-bot test Ubuntu arm Cross Checked jitstress1_tiered Build and Test
@dotnet-bot test Ubuntu arm Cross Checked r2r_jitstress1_tiered Build and Test
@dotnet-bot test Windows_NT arm Cross Checked jitstress1_tiered Build and Test
@dotnet-bot test Windows_NT arm Cross Checked r2r_jitstress1_tiered Build and Test
@dotnet-bot test Windows_NT x64_arm64_altjit Checked jitstress1_tiered
@dotnet-bot test Windows_NT x64_arm64_altjit Checked r2r_jitstress1_tiered
@dotnet-bot test CentOS7.1 x64 Checked r2r_jitstress1_tiered
@dotnet-bot test OSX10.12 x64 Checked jitstress1_tiered
@dotnet-bot test OSX10.12 x64 Checked r2r_jitstress1_tiered
@dotnet-bot test Ubuntu x64 Checked jitstress1_tiered
@dotnet-bot test Ubuntu x64 Checked r2r_jitstress1_tiered
@dotnet-bot test Windows_NT x64 Checked jitstress1_tiered
@dotnet-bot test Windows_NT x64 Checked r2r_jitstress1_tiered
@dotnet-bot test Windows_NT x86_arm_altjit Checked jitstress1_tiered
@dotnet-bot test Windows_NT x86_arm_altjit Checked r2r_jitstress1_tiered
@dotnet-bot test Windows_NT x86 Checked jitstress1_tiered
@dotnet-bot test Windows_NT x86 Checked r2r_jitstress1_tiered

The new tiered compilation jitstress2 legs
@dotnet-bot test Ubuntu arm64 Cross Checked r2r_jitstress2_tiered Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked jitstress2_tiered Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked r2r_jitstress2_tiered Build and Test
@dotnet-bot test Ubuntu arm Cross Checked jitstress2_tiered Build and Test
@dotnet-bot test Ubuntu arm Cross Checked r2r_jitstress2_tiered Build and Test
@dotnet-bot test Windows_NT arm Cross Checked jitstress2_tiered Build and Test
@dotnet-bot test Windows_NT arm Cross Checked r2r_jitstress2_tiered Build and Test
@dotnet-bot test Windows_NT x64_arm64_altjit Checked jitstress2_tiered
@dotnet-bot test Windows_NT x64_arm64_altjit Checked r2r_jitstress2_tiered
@dotnet-bot test CentOS7.1 x64 Checked r2r_jitstress2_tiered
@dotnet-bot test OSX10.12 x64 Checked jitstress2_tiered
@dotnet-bot test OSX10.12 x64 Checked r2r_jitstress2_tiered
@dotnet-bot test Ubuntu x64 Checked jitstress2_tiered
@dotnet-bot test Ubuntu x64 Checked r2r_jitstress2_tiered
@dotnet-bot test Windows_NT x64 Checked jitstress2_tiered
@dotnet-bot test Windows_NT x64 Checked r2r_jitstress2_tiered
@dotnet-bot test Windows_NT x86_arm_altjit Checked jitstress2_tiered
@dotnet-bot test Windows_NT x86_arm_altjit Checked r2r_jitstress2_tiered
@dotnet-bot test Windows_NT x86 Checked jitstress2_tiered
@dotnet-bot test Windows_NT x86 Checked r2r_jitstress2_tiered


The new 'no_tiered_compilation' explicit legs (because tiered compilation isn't on by default right now, these should be identical to existing 'normal' and 'r2r' runs)
@dotnet-bot test Windows_NT arm64 Cross Checked no_tiered_compilation Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked r2r_no_tiered_compilation Build and Test
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation Build and Test
@dotnet-bot test Ubuntu arm Cross Checked r2r_no_tiered_compilation Build and Test
@dotnet-bot test Windows_NT arm Cross Checked no_tiered_compilation Build and Test
@dotnet-bot test Windows_NT arm Cross Checked r2r_no_tiered_compilation Build and Test
@dotnet-bot test Windows_NT x64_arm64_altjit Checked no_tiered_compilation
@dotnet-bot test Windows_NT x64_arm64_altjit Checked r2r_no_tiered_compilation
@dotnet-bot test CentOS7.1 x64 Checked r2r_no_tiered_compilation
@dotnet-bot test OSX10.12 x64 Checked no_tiered_compilation
@dotnet-bot test OSX10.12 x64 Checked r2r_no_tiered_compilation
@dotnet-bot test Ubuntu x64 Checked no_tiered_compilation
@dotnet-bot test Ubuntu x64 Checked r2r_no_tiered_compilation
@dotnet-bot test Windows_NT x64 Checked no_tiered_compilation
@dotnet-bot test Windows_NT x64 Checked r2r_no_tiered_compilation
@dotnet-bot test Windows_NT x86_arm_altjit Checked no_tiered_compilation
@dotnet-bot test Windows_NT x86_arm_altjit Checked r2r_no_tiered_compilation
@dotnet-bot test Windows_NT x86 Checked no_tiered_compilation
@dotnet-bot test Windows_NT x86 Checked r2r_no_tiered_compilation

**Comment by noahfalk** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked r2r_jitstress1_tiered Build and Test

**Comment by noahfalk** [comment]
@dotnet-bot test Windows_NT arm64 Cross Checked r2r_jitstress1 Build and Test
@dotnet-bot test Windows_NT arm64 Cross Checked r2r_jitstress1_tiered Build and Test

**Comment by noahfalk** [comment]
@BruceForstall @kouvel @AndyAyersMS 
The three test legs that are being scheduled as gated PR are passing (these are all the innerloop tests, but the code has different display name conventions across architectures):
   - Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test
   - Windows_NT x64 Checked Build and Test (Jit - TieredCompilation=0)
   - Windows_NT x86 Checked Build and Test (Jit - TieredCompilation=0)

Some of the other legs continue to fail, but they are unchanged by this commit so I don't think there is any reason to hold this change because of them. On OSX there were significant infrastructure issues either with networking or with the build failing that appear independent of any tiered compilation work. On the remaining legs there are ~10 tests that fail in one or more places with tiered compilation enabled and the same tests do not fail with tiered compilation disabled. I'll work with Kount to get these triaged.

**Comment by AndyAyersMS** [comment]
A bunch of your test legs are hitting #18963 which predominantly (exclusively?) happens with minopts and hence tier0. The change that caused it has been backed out so you might rerun some of these.

**Comment by AndyAyersMS** [comment]
A few more are hitting #18777 which has likewise been cleaned up.

**Comment by noahfalk** [comment]
@BruceForstall @kouvel @AndyAyersMS - happy on this one? I'll merge later tonight if I don't hear anything.

File: netci.groovy [hunk 1]
```diff
@@ -1813,14 +1813,13 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
...
-                        //    if (configuration == 'Checked') {
-                        //        def displayStr = getStressModeDisplayName(scenario)
-                        //        Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} Build and Test (Jit - ${displayStr})")
-                        //    }
-                        //    break
+                        case 'no_tiered_compilation_innerloop':
+                            // Default trigger
+                            if (configuration == 'Checked') {
+                                def displayStr = getStressModeDisplayName(scenario)
+                                Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} Build and Test (Jit - ${displayStr})")
```
Comment (BruceForstall) [inline netci.groovy]: IMO, this should include the word "Innerloop" like the "standard" innerloop job:
```
Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} Innerloop Build and Test (Jit - ${displayStr})")
```
Same for x86, below.

**Comment by AndyAyersMS** [comment]
At this point we should prioritize switching over. We can fix up testing and tooling in parallel or more likely once we get a bit of mileage on things.

**Pull request merged**